Add release summary for SDK 1.2 (#1849)
Qiskit/documentation: docs/guides/latest-updates.mdx

{/* release summaries written by product owners */}

### Qiskit SDK v1.2.0

*To see the release notes for all versions, visit the [Qiskit SDK release notes](/api/qiskit/release-notes).*

> Qiskit 1.2 (stable release) is now available on [PyPI](https://pypi.org/project/qiskit/1.2.0/) and [GitHub](https://github.com/Qiskit/qiskit/releases/tag/1.2.0). Thanks to the whole dev team and all external contributors!

Reviewer: Nits:

1. Not sure `whole` is bringing much value?
2. Probably better to call it `open-source` rather than `external`

```suggestion
Qiskit 1.2 (stable release) is now available on [PyPI](https://pypi.org/project/qiskit/1.2.0/) and [GitHub](https://github.com/Qiskit/qiskit/releases/tag/1.2.0). Thanks to the maintainers and open-source contributors!
```

- Many of the core structures have been moved to Rust, allowing for significant speedups mostly in the construction, manipulation (including copy and parameter assignment), and synthesis of circuits.
- Better compiler quality due to the addition of unitary peephole optimization (for optimization levels 2 and 3) and improvements in Sabre.
- [`generate_preset_pass_manager`](/api/qiskit/transpiler_preset#generate_preset_pass_manager) importance highlight, which is now importable from the top level.

Important deprecations:

> - Primitive V1 reference implementations (like `qiskit.primitives.Estimator`) and tooling are deprecated in favor of V2 Primitives. The Primitive V1 base classes remain available.

Author: ```suggestion
- Primitive V1 reference implementations (like `qiskit.primitives.Estimator`) and tooling are deprecated in favor of [V2 primitives](/api/qiskit/primitives). The primitive V1 base classes remain available.
```

> - Primitive V1 reference implementations (like `qiskit.primitives.Estimator`) and tooling are deprecated in favor of V2 Primitives. The Primitive V1 base classes remain available.

Reviewer: Should this link to https://docs.quantum.ibm.com/api/qiskit/primitives#overview-of-estimatorv2?

Author: Wouldn't we also need to point to Samplerv2? How about the level-up https://docs.quantum.ibm.com/api/qiskit/primitives instead

> - Backend V1 is now deprecated in favor of [`BackendV2`](/api/qiskit/qiskit.providers.BackendV2), including the deprecation of `Qobj`, `assemble`, and other scaffolding related to the old V1.

Reviewer: ```suggestion
- [`BackendV1`](/api/qiskit/qiskit.providers.BackendV1) is now deprecated in favor of [`BackendV2`](/api/qiskit/qiskit.providers.BackendV2), including the deprecation of `Qobj`, `assemble`, and other scaffolding related to the old V1.
```

Author: Is it ok to point to something that's deprecated, especially if we're not also linking to the pages for other deprecations `Qobj` and `assemble`?

Reviewer: Yeah because this page won't live forever, so it's fine to link to it. I think it can be helpful so that people can see what the old interface was.

> especially if we're not also linking to the pages for other deprecations Qobj and assemble?

fwiw, those could also be linked to

> - Support for Python 3.8 will be dropped in the next release, 1.3. Python 3.9 will then be the new minimal required version.

Reviewer: ```suggestion
- Support for Python 3.8 will be dropped in the next release, v1.3. Python 3.9 will then be the new minimal required version.
```

Remember, because this a minor release, it will not introduce breaking changes for users moving from Qiskit SDK v1.0. Any deprecations announced with this release refer only to things that will now start throwing deprecation warnings. None of these will be removed until the release of Qiskit v2.0, pursuant to Qiskit’s official [deprecation policy](https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md#).

> For more details, be sure to check [the v1.2 release notes](/api/qiskit/release-notes/1.2).

Reviewer 2: ```suggestion
For more details, be sure to check [the v1.2 release notes](/api/qiskit/release-notes/1.2).
```
```suggestion
For more details, check [the v1.2 release notes](/api/qiskit/release-notes/1.2).
```
